codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java

```diff
@@ -587,7 +591,7 @@ private State readHeaders(ByteBuf buffer) {
 
         State nextState;
 
-        if (isContentAlwaysEmpty(message)) {
+        if (isContentAlwaysEmpty(message) || HttpUtil.hasEmptyContent(message)) {
```

Comment:
And if you meant that this does require content-length to be set, it doesn't. The routine getContentLength (which I didn't write) handles this.